src/formatDate.js

```diff
@@ -50,7 +50,70 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const arrDate = date.split(oldSeparator);
+
+  const result = [];
+  let year, month, day, oldYearFormat;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case ('YY'):
+        oldYearFormat = 2;
+        year = arrDate[i];
+        break;
+
+      case ('YYYY'):
+        oldYearFormat = 4;
+        year = arrDate[i];
+        break;
+
+      case ('MM'):
+        month = arrDate[i];
+        break;
+
+      case ('DD'):
+        day = arrDate[i];
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
```

Comment:
It's better to use the map() method to create the result array, which simplifies the code and makes it more readable.
```
const result = toFormat.map((format) => {...